File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/FilterInteractor.java
package com.borisruzanov.russianwives.mvp.model.interactor;

import com.borisruzanov.russianwives.models.SearchModel;
import com.borisruzanov.russianwives.mvp.model.repository.FilterRepository;

import java.util.List;

public class FilterInteractor {

    private FilterRepository filterRepository;

    public FilterInteractor(FilterRepository filterRepository) {
        this.filterRepository = filterRepository;
    }

    public List<String> getPrefsValues(){
        return filterRepository.getPrefsValues();
    }

    public void setPrefsValues(List<SearchModel> searchModels){
        filterRepository.setPrefsValues(searchModels);
    }

    /*public String setValue(String key){
        return filterRepository.setValue(key);
    }

    public void setValue(String key, String value){
        filterRepository.setValue(key, value);
    }*/

}

File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/filter/FilterInteractor.kt
package com.borisruzanov.russianwives.mvp.model.interactor.filter

import com.borisruzanov.russianwives.models.SearchModel
import com.borisruzanov.russianwives.mvp.model.repository.filter.FilterRepository
import javax.inject.Inject

class FilterInteractor @Inject constructor(private val filterRepository: FilterRepository) {

    val prefsValues: List<String>
        get() = filterRepository.prefsValues

    fun setPrefsValues(searchModels: List<SearchModel>) {
        filterRepository.setPrefsValues(searchModels)
    }

    /*public String setValue(String key){
        return filterRepository.setValue(key);
    }

    public void setValue(String key, String value){
        filterRepository.setValue(key, value);
    }*/

}
